### PR TITLE
Updage to Angular 14, TypeScript 4.6 and RxJS 7.5

### PR DESCRIPTION
**Pull request by fjaguado**
# PR Details

Update to Angular 14 and TypeScript 4.6
Fix some builder issues while updating RxJS to 7.5

## Description

Followed Angular Update Guide

The update to version 13 has been done first
https://update.angular.io/?l=3&v=12.0-13.0 but adding @angular-eslint/schematics@13 to the ng update

And then to version 14 following the same pattern
https://update.angular.io/?l=3&v=13.0-14.0 but adding @angular-eslint/schematics@14 to the ng update

At this point the following libraries have been updated or the version are fixed: 

- "eslint-plugin-import": "2.26.0"
- "eslint-plugin-jsdoc": "39.3.6"
- "eslint-plugin-prefer-arrow": "1.2.3"
- "karma-jasmine-html-reporter": "1.5.4"

The following packages have been removed because they are deprecated:

- "@angular-devkit/build-ng-packagr": "^0.1002.0" 

Instead of build-ng-packagr we moved to build-angular and changed a line in angular.json

- "builder": "@angular-devkit/build-angular:ng-packagr"
Also the Travis configuration has been updated

## Related Issue
#76 

## Motivation and Context

This update is part of a cybersecurity demand on how update a full system can be achieved. Also, we want to have the latest technology too.

## How has this been tested

It has been executed tests

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the CONTRIBUTING document
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A new branch needs to be created from master to evolve previous versions
- [x] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [x] Add the issue into the right [project](https://github.com/systelab/systelab-translate/projects) with the proper status (In progress)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,58 +4,57 @@
...
-    "@angular/router": "^12.1.1",
+    "@angular/animations": "^14.2.3",
+    "@angular/common": "^14.2.3",
+    "@angular/compiler": "^14.2.3",
+    "@angular/core": "^14.2.3",
+    "@angular/forms": "^14.2.3",
+    "@angular/platform-browser": "^14.2.3",
+    "@angular/platform-browser-dynamic": "^14.2.3",
+    "@angular/router": "^14.2.3",
     "@ngx-translate/core": "^13.0.0",
```
Comment (OscarDeDios) [inline package.json:22]: There is a new ngx-translate 14.0 version

File: package.json [hunk 2]
```diff
@@ -4,58 +4,57 @@
...
     "@ngx-translate/http-loader": "^6.0.0",
```
Comment (OscarDeDios) [inline package.json:23]: There is a new 7.0 version

File: package.json (hunk 1 above)
Comment (fjaguado) [reply package.json:22]: Yes, I know, but this is not the scope of this update. These change will be done in the future.

File: package.json (hunk 2 above)
Comment (fjaguado) [reply package.json:23]: Yes, I know, but this is not the scope of this update. These change will be done in the future.

**Review by OscarDeDios — APPROVED** [review]

**Review by carlosra85 — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,16 @@
 sudo: required
-dist: trusty
+dist: jammy
 language: node_js
 node_js:
-  - '12'
+  - '14.20.1'
```
Comment (joaquimvila) [inline .travis.yml:5]: Why using NodeJS 14.x on continuous integration and then using version 16.x on file for node version manager? I would go also for the 16.x version on TravisCI considering this is the latest LTS.

Comment (fjaguado) [reply .travis.yml:5]: Because of a problem with Systelab-Preferences I prefered to use the same version for all Systelab libraries

**Pull request merged**